### PR TITLE
Added a few mappings in ChunkGeneratorType now #1384 is closed

### PR DESCRIPTION
**Pull request by Kroppeb**

### CODE REVIEW

File: mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping [hunk 1]
```diff
@@ -2,16 +2,31 @@ CLASS net/minecraft/class_5284 net/minecraft/world/gen/chunk/ChunkGeneratorType
 	FIELD field_24514 defaultBlock Lnet/minecraft/class_2680;
 	FIELD field_24515 defaultFluid Lnet/minecraft/class_2680;
 	FIELD field_24516 config Lnet/minecraft/class_5311;
+	FIELD field_24780 OBJECT_CODEC Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping]: the codec for plain chunk generator type, without string shortcuts? imo the one that does not accept shortcuts should be `CODEC`, and one that accept alternatives as `PRESET_CODEC` etc

File: mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping [hunk 2]
```diff
@@ -2,16 +2,31 @@ CLASS net/minecraft/class_5284 net/minecraft/world/gen/chunk/ChunkGeneratorType
...
+	FIELD field_24781 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_24782 noiseConfig Lnet/minecraft/class_5309;
 	FIELD field_24783 bedrockCeilingY I
 	FIELD field_24784 bedrockFloorY I
+	FIELD field_24785 seaLevel I
+	FIELD field_24786 disabledMobGeneration Z
```
Comment (liach) [inline mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping:11]: maybe mob spawn than generation, so `noMobSpawn`?

File: mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping (hunk 1 above)
Comment (Kroppeb) [reply mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping]: Should I also rename `ChunkGeneratorType.Preset.PRESET_CODEC` to `CODEC` instead?

Comment (liach) [reply mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping]: wait, so what does this codec do 🤔

Comment (liach) [reply mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping]: The preset's own codec is the only codec so it can be named `CODEC`

Comment (Kroppeb) [reply mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping]: Yeah, I hadn't noticed they weren't strictly in the same class yet

File: mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping (hunk 2 above)
Comment (Kroppeb) [reply mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping:11]: mmh, the string is `"disable_mob_generation"`

File: mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping [hunk 3]
```diff
@@ -22,8 +37,12 @@ CLASS net/minecraft/class_5284 net/minecraft/world/gen/chunk/ChunkGeneratorType
 		COMMENT <p>If a number greater than 255 is returned, the floor will not be generated.
 	METHOD method_28005 getDefaultBlock ()Lnet/minecraft/class_2680;
 	METHOD method_28006 getDefaultFluid ()Lnet/minecraft/class_2680;
-	METHOD method_28007 getConfig ()Lnet/minecraft/class_5311;
+	METHOD method_28007 getStructuresConfig ()Lnet/minecraft/class_5311;
+	METHOD method_28559 getNoiseConfig ()Lnet/minecraft/class_5309;
+	METHOD method_28561 getSeaLevel ()I
+	METHOD method_28562 disabledMobGeneration ()Z
```
Comment (liach) [inline mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping:43]:
```suggestion
	METHOD method_28562 disablesMobGeneration ()Z
```

Comment (YanisBft) [reply mappings/net/minecraft/world/gen/chunk/ChunkGeneratorType.mapping:43]: More `isMobGenerationDisabled`

**Comment by YanisBft** [comment]
Superseded by #1628

**Pull request closed without merging**